File: search_algorithms/binary_search.py
# Implementation of Binary Search Algorithm

def binary_search(L, x):
    """Given a sorted list L and a value x, find the index of x in the list L."""
    low = 0
    high = len(L) - 1

    while low <= high:
        mid = (low + high) // 2
        if L[mid] == x:
            return mid
        elif L[mid] < x:
            low = mid + 1
        else:
            high = mid - 1

    return -1


if __name__ == "__main__":
    L = [1, 2, 3, 4, 7, 10, 12, 37, 42, 99]
    x = 37
    print(binary_search(L, x))

    

File: search_algorithms/binary_search.c
// Implementation of Binary Search Algorithm

#include <stdio.h>


int binary_search(int arr[], int n, int target)
{
    int low = 0;
    int high = n-1;

    while (low <= high)
    {
        int mid = low + (high - low) / 2;

        if (arr[mid] == target)
        {
            return mid;
        }
        if (arr[mid] < target)
        {
            low = mid + 1;
        }
        else 
        {
            high = mid - 1;
        }
    }
    return -1;
}

int main()
{
    int arr[] = {44, 23, 192, 223, 513, 25};
    int n = sizeof(arr)/sizeof(arr[0]);

    int target = 513;
    printf("Element %d found at index %d\n", target, binary_search(arr, n, target));
}
